### ISSUE TITLE
Issue with $ref in model properties for v1.2

### ISSUE THREAD

**Issue by cancan101** [swagger-api/swagger-js#420]
There seems to be an issue with the spec converter handling `$ref` in swagger 1.2 models.

See: https://gist.github.com/cancan101/79ecd0e07601da6c7cae

I think something like this is missing:
```diff
SwaggerSpecConverter.prototype.models = function(obj, swagger) {
       if(existingProperty['enum']) {
         property['enum'] = existingProperty['enum'];
       }
+      if(existingProperty['$ref']) {
+        property['$ref'] = existingProperty['$ref'];
+      }
       if(typeof existingProperty.required === 'boolean' && existingProperty.required === true) {
         _enum.push(propertyName);
       }
```
Which leads to:
![image](https://cloud.githubusercontent.com/assets/51059/7558478/49bd8676-f773-11e4-932c-358cd56d2130.png)
rather than:
![image](https://cloud.githubusercontent.com/assets/51059/7558481/5d89a7ac-f773-11e4-875d-9161a4b70d51.png)

**Comment by webron** [comment]
Seems like just a tiny step before submitting a PR (against the develop_2.0 branch of course).

Out of curiosity, what generates that 1.2 file? It's not 100% compliant with the spec.

**Comment by cancan101** [comment]
A slightly modified django rest swagger though I don't think the modifications should have made it less compliant.

**Comment by webron** [comment]
Perhaps the original wasn't 100% compliant. The model properties have a few properties of their own which are either not supported or contain invalid values. This is unrelated to the problem and should not 'explode' the client anyways.

**Comment by cancan101** [comment]
Would you be okay with the following which does allow for a more lenient interpretation of the 1.2 spec: 
https://github.com/cancan101/swagger-js/commit/0945c776f54415d7937fab390f279e89534d41e4 ?

**Comment by webron** [comment]
I'll leave this one to @fehguy ;)

**Comment by cancan101** [comment]
Referenced commit also fixes: #424

**Comment by fehguy** [comment]
I just put your sample spec from [here](https://gist.githubusercontent.com/cancan101/79ecd0e07601da6c7cae/raw/928b887a44da9b80c28509c7c54fb2c82975e185/$ref-issue.json) into the UI and get what you'd expect.  Did that gist change?

![image](https://cloud.githubusercontent.com/assets/249413/7609114/c402419e-f93d-11e4-8d95-ba61a7eb138f.png)

**Comment by fehguy** [comment]
Fixed in 2.1.5-M2
